src/components/feedback/feedbackModal.tsx

```diff
@@ -1,10 +1,121 @@
-import React, {FormEvent, useEffect, useRef} from 'react';
-import {css} from '@emotion/react';
+import React, {Fragment, useEffect, useRef, useState} from 'react';
 import styled from '@emotion/styled';
 
 import {useFocusTrap} from '../hooks/useFocusTrap';
 import {useShortcut} from '../hooks/useShortcut';
 
+import {FeedbackForm} from './feedbackForm';
+import {FeedbackSuccessMessage} from './feedbackSuccessMessage';
+import {sendFeedbackRequest} from './sendFeedbackRequest';
+
+interface RenderFunctionProps {
+  /**
+   * Is the modal open/visible
+   */
+  open: boolean;
+
+  /**
+   * Shows the feedback modal
+   */
+  showModal: () => void;
+}
+type FeedbackRenderFunction = (
+  renderFunctionProps: RenderFunctionProps
+) => React.ReactNode;
+
+interface FeedbackModalProps {
+  children: FeedbackRenderFunction;
+  title: string;
+}
+
+interface FeedbackFormData {
+  comment: string;
+  email: string;
+  name: string;
+}
+
+async function sendFeedback(
+  data: FeedbackFormData,
+  replayId: string,
+  pageUrl: string
+): Promise<Response | null> {
+  const feedback = {
+    message: data.comment,
+    email: data.email,
+    replay_id: replayId,
+    url: pageUrl,
+  };
+  return await sendFeedbackRequest(feedback);
+}
+
+function stopPropagation(e: React.MouseEvent) {
+  e.stopPropagation();
+}
+
+export function FeedbackModal({title, children}: FeedbackModalProps) {
+  const [open, setOpen] = useState(false);
+  const dialogRef = useRef<HTMLDialogElement>(null);
+  const [showSuccessMessage, setShowSuccessMessage] = useState(false);
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleSubmit = (data: FeedbackFormData) => {
+    const replay = window.Sentry?.getCurrentHub?.()
+      ?.getClient()
+      ?.getIntegration(window.Sentry?.Replay);
+
+    // Prepare session replay
+    replay?.flush();
+    const replayId = replay?.getReplayId();
+
+    const pageUrl = document.location.href;
+
+    sendFeedback(data, replayId, pageUrl).then(response => {
+      if (response) {
+        setOpen(false);
+        setShowSuccessMessage(true);
+      } else {
+        // eslint-disable-next-line no-alert
+        alert('Error submitting your feedback. Please try again');
+      }
+    });
+  };
+
+  const showModal = () => {
+    setOpen(true);
+  };
+
+  useEffect(() => {
+    if (!showSuccessMessage) {
+      return () => {};
+    }
+    const timeout = setTimeout(() => {
+      setShowSuccessMessage(false);
+    }, 6000);
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [showSuccessMessage]);
+
+  useFocusTrap(dialogRef, open);
+  useShortcut('Escape', handleClose);
+
+  return (
+    <Fragment>
+      <Dialog id="feedbackModal" open={open} ref={dialogRef} onClick={handleClose}>
+        <Content onClick={stopPropagation}>
+          <Header>{title}</Header>
+          {open && <FeedbackForm onSubmit={handleSubmit} onClose={handleClose} />}
```

Comment:
I opted to not render the form when it's not visible as to also kill the form state when the modal is closed (there was some useEffect/ref shenanigans going on from the hackweek project).

I had Header in Form initially, but moved into into modal to avoid passing `title` to another component